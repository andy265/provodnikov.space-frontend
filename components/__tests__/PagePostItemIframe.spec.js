const { shallow } = require('vue-test-utils')
const { createRenderer } = require('vue-server-renderer')
jest.mock('~/components/HasModelMixin', () => require('HasModelMixin'))
const PagePostItemIframe = require('~/components/PagePostItemIframe')

describe('PagePostItemIframe.vue', () => {
  it('has same HTML structure when state is "index"', () => {
    let Component = Object.assign({}, PagePostItemIframe, { components: {} })
    let wrapper = shallow(Component, {
      propsData: {
        state: 'index',
        isPreview: false,
        postUrl: 'http://localhost/post/post_url',
        value: {
          id: 'ak8kej76JQY',
          width: 560,
          height: 315
        }
      }
    })
    let renderer = createRenderer()
    // eslint-disable-next-line
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })
})
