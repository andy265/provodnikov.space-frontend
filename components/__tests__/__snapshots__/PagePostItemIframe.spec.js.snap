// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagePostItemIframe.vue has same HTML structure when state is "index" 1`] = `
<div class="page-post-item-iframe">
    <AppIframe src="https://www.youtube.com/embed/ak8kej76JQY?rel=0&amp;amp;showinfo=0" width="560" height="315"></AppIframe>
    <!---->
</div>
`;
